effect/shared/src/main/scala/scalaz/effect/IO.scala

```diff
@@ -1,16 +1,20 @@
-// Copyright (C) 2017 John A. De Goes. All rights reserved.
+// Copyright (C) 2017-2018 John A. De Goes. All rights reserved.
 package scalaz.effect
 
 import scala.annotation.switch
-import scala.concurrent.duration.Duration
+import scala.concurrent.duration._
 
+import scalaz.data.Disjunction
 import scalaz.data.Disjunction._
 import scalaz.data.Maybe
+import scalaz.Void
+
+import scalaz.effect.Errors._
 
 /**
- * An `IO[A]` ("Eye-Oh of A") is an immutable data structure that describes an
- * effectful action that may throw an exception (`Throwable`), run forever, or
- * produce a single `A` at some point in the future.
+ * An `IO[E, A]` ("Eye-Oh of A") is an immutable data structure that describes an
+ * effectful action that may fail with an `E`, run forever, or produce a single
+ * `A` at some point in the future.
```

Comment:
Looking forward to more documentation on if/when this guarantee fails. Quoting what I asked on [Gitter](https://gitter.im/scalaz/scalaz?at=5ace33ff270d7d3708b80e16):

> does anybody plan to state "formally" under what conditions you get other exceptions? Not sure Scalazzi is enough
> it seems E behaves basically like a phantom type parameter at the edges
> 1. so you need to import Java functions with the correct spec
> 2. you have "async. exceptions" from the runtime, you need to consider them
> 3. if you listen to Java checked exception specs, you need to consider unchecked exceptions as "async"